examples/passport-login/src/controllers/oauth2.controller.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/example-passport-login
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  get,
+  RestBindings,
+  Response,
+  RequestWithSession,
+  param,
+} from '@loopback/rest';
+import {authenticate} from '@loopback/authentication';
+import {inject} from '@loopback/core';
+import {SecurityBindings, UserProfile} from '@loopback/security';
+
+/**
+ * Login controller for third party oauth provider
+ */
+export class Oauth2Controller {
+  constructor() {}
+
+  @authenticate('oauth2')
+  @get('/auth/thirdparty/{provider}')
+  /**
+   * Endpoint: '/auth/thirdparty/{provider}'
+   *          an endpoint for api clients to login via a third party app, redirects to third party app
+   */
+  loginToThirdParty(
+    @param.path.string('provider') provider: string,
+    @inject('authentication.redirect.url')
```

Comment:
a reminder: doesn't have to apply in this PR, please create binding keys for the redirect url and status.

Reply:
agreed